pkg/server/auth/passthrough.go

```diff
@@ -58,5 +57,44 @@ func (pg *BearerTokenPassthroughPrincipalGetter) Principal(r *http.Request) (*Us
 		return nil, fmt.Errorf("error: user token authentication failed")
 	}
 
-	return &UserPrincipal{Token: token}, nil
+	return NewUserPrincipal(Token(token)), nil
+}
+
+// NewJWTPassthroughCookiePrincipalGetter creates and returns a new
+// JWTPassthroughCookiePrincipalGetter.
+func NewJWTPassthroughCookiePrincipalGetter(log logr.Logger, verifier *oidc.IDTokenVerifier, cookieName string) PrincipalGetter {
+	return &JWTPassthroughCookiePrincipalGetter{
+		log:        log,
+		verifier:   verifier,
+		cookieName: cookieName,
+	}
+}
+
+// JWTPassthroughCookiePrincipalGetter inspects a cookie for a JWT token and returns a
+// principal value.
+//
+// The JWT Token is parsed, and the token and user/groups are available.
+type JWTPassthroughCookiePrincipalGetter struct {
+	log        logr.Logger
+	verifier   *oidc.IDTokenVerifier
+	cookieName string
+}
+
+// Principal implements the PrincipalGetter by pasing the cookie, and if it's
+// valid, it stores the token and user details on the principal.
+func (pg *JWTPassthroughCookiePrincipalGetter) Principal(r *http.Request) (*UserPrincipal, error) {
+	cookie, err := r.Cookie(pg.cookieName)
+	if err == http.ErrNoCookie {
+		return nil, nil
+	}
+
+	principal, err := parseJWTToken(r.Context(), pg.verifier, cookie.Value)
+	if err != nil {
+		return nil, fmt.Errorf("failed to parse for passthrough: %w", err)
+	}
+
+	pg.log.V(4).Info("passing through token")
```

Comment:
I can never remember which direction the verbosity numbers work in logr. This is a "tracing-ish" verbosity level, right? So it's not supposed to show up?

Reply:
Yeah, the numbers reflect "verbosity" for things that accept `-v=0` style command-line options.

I _think_ it's probably derived from older commands that had `-vvv` for "more verbose" as cmd-line options.

And yes, that's basically not supposed to show up, unless you really want it to, because it would be logged on each request.